pendant/device.ts

```diff
@@ -64,63 +65,34 @@ export class PendantDevice extends EventEmitter {
         await this.control(packet);
     }
 
-    public async init() {
+    public init() {
```

Comment:
This function had a purpose. On some OSs (I forgot which, but I tested this to work on Windows, Linux and macOS), this will only be 1 device, not 2, hence trying both of them to figure out which one is actually the one we need. (And I wouldn't trust the `sort by path` thing too much, either)

Reply:
Ah, thanks for pointing this out.  The batch file led me to think this was meant to be Windows-only (I only use it on Windows).  Let me have a think about how best to make the new functionality cross-platform.

Reply:
P.S. My motivation is that I want to start carvera-pendant at boot time and just have everything work when I use the Carvera and the pendant later on.  (I only turn the pendant on when I need it.)

Reply:
I see. I also run everything at boot time (on a small Linux laptop, controller via wine), but I never unplug my pendant, so that makes sense.

Absolutely happy for improvements, though :) I just initially only made it for my own use-cases

Reply:
I've just tested my version on Linux (Ubuntu 22.04 LTS) and it worked out of the box.  Let me see if I can find a way to test it on OSX.

Reply:
I've refactored the init logic in a [separate commit](https://github.com/Doridian/carvera-pendant/commit/8e10a5357f2e73c115c8d923bc6e6607db79b925) to cleanly handle platform differences.

It works on Windows and Linux.

Unfortunately, there's something about my OSX setup that prevents **both the original** (a clean pull from your repo) and this new version from working.  The HID device shows up on `hidutil` and on `node-hid`'s `HID.devices()`, but trying to open it fails with "cannot open device with path ...".  I tried opening by VID:PID but that fails with a similar error.

I tried a few things like running as root and making sure the device doesn't claim to be a keyboard or a mouse (it doesn't: its usage page is 0xFF00, i.e. vendor-specific).

I've run out of things to try at this point.

I don't suppose you might have a working OSX setup on which to test this?

Reply:
@zvea Well if the existing logic is already not currently working on macOS then we can merge this change right now and I'll log an issue ticket and see if I can fix it when I can plug the pendant into my MacBook to test. Sound good?

Reply:
Sounds good, thanks.